classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,378 @@
+				<p><?php printf( __( 'Could not execute the action: %s', 'action-scheduler' ), $action_hook_html ); ?></p>
+			</div>
+			<?php endif;
+		}
+	}
+
+
+	/**
+	 * Prints the scheduled date in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 */
+	public function column_scheduled( $row ) {
+		if ( $row['scheduled']->next() ) {
+			$next_timestamp = $row['scheduled']->next()->format( 'U' );
+			echo $row['scheduled']->next()->format( 'Y-m-d H:i:s' );
+			if ( gmdate( 'U' ) > $next_timestamp ) {
+				printf( __( ' (%s ago)', 'action-scheduler' ), human_time_diff( gmdate( 'U' ), $next_timestamp ) );
+			} else {
+				echo ' (' . human_time_diff( gmdate( 'U' ), $next_timestamp ) . ')';
+			}
+		}
+	}
+
+	/**
+	 * Bulk delete
+	 *
+	 * Deletes actions based on their ID. This is the handler for the bulk delete. It assumes the data
+	 * properly validated by the callee and it will delete the actions without any extra validation.
+	 *
+	 * @param array $ids
+	 *
+	 * @return void
+	 */
+	protected function bulk_delete( array $ids, $ids_sql ) {
+		foreach ( $ids as $id ) {
+			$this->store->delete_action( $id );
+		}
+	}
+
+	/**
+	 * Implements the logic behind running an action. PP_Table_List validates the request and their
+	 * parameters are valid.
+	 */
+	protected function row_action_run( $action_id ) {
+		try {
+			ActionScheduler::runner()->process_action( $action_id );
+			$success = 1;
+		} catch ( Exception $e ) {
+			$success = 0;
+		}
+
+		set_transient( 'actionscheduler_admin_executed', compact( 'action_id', 'success' ), 30 );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function prepare_items() {
+		$this->process_bulk_action();
+
+		$this->process_row_actions();
+
+		if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			// _wp_http_referer is used only on bulk actions, we remove it to keep the $_GET shorter
+			wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
+			exit;
+		}
+
+		$this->prepare_column_headers();
+
+		$per_page = $this->get_items_per_page( $this->package . '_items_per_page', $this->items_per_page );
+		$query = array(
+			'per_page' => $per_page,
+			'offset'   => $this->get_items_offset(),
+			'status'   => $this->get_request_status(),
+			'orderby'  => 'modified',
+			'order'    => 'ASC',
+		);
+
+		$this->items = array();
+
+		$total_items = $this->store->query_actions_count( $query );
+
+		foreach ( $this->store->query_actions( $query ) as $id ) {
+			$action = $this->store->fetch_action( $id );
+			$this->items[ $id ] = array(
+				'ID'     => $id,
+				'hook'   => $action->get_hook(),
+				'status' => $action->get_status(),
+				'args'   => $action->get_args(),
+				'group'  => $action->get_group(),
+				'log'    => $this->logger->get_logs( $id ),
+				'recurrence' => $this->get_recurrence( $action ),
+				'scheduled'  => $action->get_schedule(),
+			);
+		}
```

Comment:
Agreed. I also considered `$action` being the "item" instead of looping over all the items here and determining half the data. I'm not sure if this should be blocking, but it could definitely use improvements.